setup.py

```diff
@@ -59,57 +58,41 @@ def run(self, *arg, **kw):
 with open('README.rst') as f:
     readme = f.read()
 
-install_reqs = [
+installReqs = [
     'bcrypt',
     'boto3',
     # CherryPy version is restricted due to a bug in versions >=11.1
     # https://github.com/cherrypy/cherrypy/issues/1662
     'CherryPy<11.1',
     'click',
     'filelock',
+    'funcsigs ; python_version < \'3.5\'',
     'jsonschema',
     'Mako',
     'pymongo>=3.5',
     'PyYAML',
-    'requests',
     'psutil',
     'python-dateutil',
     'pytz',
-    'six>=1.9'
+    'requests',
+    'shutilwhich ; python_version < \'3.3\'',
+    'six>=1.9',
 ]
 
-extras_reqs = {
-    'celery_jobs': ['celery'],
-    'dicom_viewer': ['pydicom'],
-    'geospatial': ['geojson'],
-    'item_tasks': ['ctk-cli'],
-    'ldap': ['pyldap'],
-    'thumbnails': ['Pillow', 'pydicom', 'numpy'],
-    'worker': ['celery>=4.0.0']
-}
-all_extra_reqs = itertools.chain.from_iterable(extras_reqs.values())
-extras_reqs['plugins'] = list(set(all_extra_reqs))
-
-if sys.version_info[0] == 2:
-    install_reqs.append('shutilwhich')
-    extras_reqs.update({
-        'hdfs_assetstore': ['snakebite'],
-        'metadata_extractor': [
-            'hachoir-core',
-            'hachoir-metadata',
-            'hachoir-parser'
-        ],
-        'plugins': extras_reqs['plugins'] + [
-            'snakebite',
-            'hachoir-core',
-            'hachoir-metadata',
-            'hachoir-parser'
-        ]
-    })
-if sys.version_info[0:2] < (3, 5):
-    install_reqs.append('funcsigs')
+extrasReqs = {}
+for pluginName in os.listdir('plugins'):
```

Comment:
This may not do the right thing when there are external plugins.  Currently, if an external plugin has a `setup.py` file *and* a `requirements.txt` file, when you ask girder to install the plugin, it runs `pip install -e {pluginPath}`.  If there isn't a `setup.py` file, then it installs `pip install -r requirements.txt`.  Now, if you run `pip install .[plugins]`, this will install the `requirements.txt` file, whereas before it would have not installed the external plugin.  Worse, if those external plugins have conflicting pinned requirements, this will fail.

Reply:
I think we should exclude external plugins entirely. We already have the established `girder-install plugin` command, and external plugins (which could disappear at any time) don't really belong as an extras option of core.

Reply:
That would be fine (but, obviously, `listdir` is an insufficient enumeration technique).

Reply:
Yeah, I think I'll use `.gitignore`. I just need to verify that it still works once it gets packaged.